### ISSUE TITLE
Update: Python with lowest minor version for each major version

### ISSUE THREAD

**Issue by cmarqu** [actions/virtual-environments#2253]
**Tool information**
- Tool name: Python
- Tool license: PSF
- Add or update? Update
- Desired version: 3.6.0, 3.7.0
- Approximate size: 

**Area for Triage**:

?

**Question, Bug, or Feature?**:
Feature

**Virtual environments affected**
- [ ] Ubuntu 16.04
- [ ] Ubuntu 18.04
- [x] Ubuntu 20.04
- [ ] macOS 10.15
- [ ] macOS 11.0
- [ ] Windows Server 2016 R2
- [ ] Windows Server 2019

**Can this tool be installed during the build?**
?

**Tool installation time in runtime**
1 min

**Are you willing to submit a PR?**
yes

Our project (and I guess many others) are specifying supported Python version like "3.6+" or ">3.6". This can  be taken to mean that version 3.6.0 is supported (i.e. the lowest minor version), and thus we would like to test this exact version.
In the past we already had breakage with older Python minor versions that we didn't find in CI.

Our project would be fine with having the listed version only available for Ubuntu 20.04, but more wouldn't hurt of course.

**Comment by maxim-lobanov** [comment]
Hello @cmarqu, 3.9.0 and 3.8.0 are already available for actions/setup-python task. So looks like your request is adding 3.6.0 and 3.7.0.
I think we can try to build these 2 versions and provide them for actions/setup-python if there is no any build issues with them (we build Python from source code and building of some versions can be tricky. Since 3.6.x and 3.7.x are built without problems, 3.6.0 and 3.7.0 should be too, I hope 😄 )

@MaksimZhukov could you please take a look when you have a chance?

**Comment by cmarqu** [comment]
> Hello @cmarqu, 3.9.0 and 3.8.0 are already available for actions/setup-python task. So looks like your request is adding 3.6.0 and 3.7.0.

Ah yes, thanks, I have updated my initial comment accordingly.

**Comment by MaksimZhukov** [comment]
Hello @cmarqu !

Unfortunately, we faced issues during building Python 3.6.0 and 3.7.0 on Ubuntu 20.04. Both 3.6.0 and 3.7.0 don’t support Ubuntu 20.04 and fail to build due to issues with openssl:
1) Python 3.7.0 on Ubuntu 20.04
```
ssl.SSLError: [SSL: CA_MD_TOO_WEAK] ca md too weak (_ssl.c:3717)
```
2) Python 3.6.0 on Ubuntu 20.04
```
ssl.SSLError: [SSL: EE_KEY_TOO_SMALL] ee key too small (_ssl.c:3288)
```
It looks like these are similar issues that are related to strict OpenSSL crypto policy. We have found two issues on the bugs.python.org that can provide more context about these errors:
1) [Issue41183](https://bugs.python.org/issue41183)
2) [issue34542](https://bugs.python.org/issue34542)

The fix for those issues has been integrated to 3.6 and 3.7, but 3.6.0 and 3.7.0 source code does not contain this fix.

We have the following ways:
1) Build the first version 3.7.1 that support Ubuntu 20.04 (Python 3.6.7 with fix for this issue is already available for actions/setup-python action). 
2) Try to suppress these errors - not sure that it won’t cause unpredictable issues later.
3) Build only 3.7.0 for Ubuntu 18.04 (build with Python 3.6.0 fails on Ubuntu 18.04 due to a different issue).

**Comment by cmarqu** [comment]
Hi @MaksimZhukov,

thanks for the thorough investigation!
I think we would be fine with only having 3.7.1 - having such low versions is not a must for us, and 3.6 usage is likely declining now.

**Comment by MaksimZhukov** [comment]
We've published Python 3.7.1 release for all images to the `actions/python-versions` repository in scope of [this pull request](https://github.com/actions/python-versions/pull/70). This Python version can be installed in run-time by [actions/setup-python](https://github.com/actions/setup-python) action:
```
jobs:
  setup-python:
    name: Setup ${{ matrix.python }} ${{ matrix.os }}
    runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [macos-latest, windows-latest, ubuntu-16.04, ubuntu-18.04, ubuntu-20.04]
        python: ['3.7.1']
    steps:
    - name: setup-python ${{ matrix.python }}
      uses: actions/setup-python@v2
      with:
        python-version: ${{ matrix.python }}
```

@cmarqu Could you please clarify if it works for you?

**Comment by cmarqu** [comment]
@MaksimZhukov Thanks, I tested with `ubuntu-20.04`, `macos-latest` and `windows-latest` and it works fine.
Great support!
